core-bundle/tests/Search/Backend/Provider/TableDataContainerProviderTest.php

```diff
@@ -0,0 +1,135 @@
+            $this->mockContaoFramework(),
+            $this->createMock(ResourceFinder::class),
+            $this->createMock(Connection::class),
+            $this->createMock(RecordLabeler::class),
+            $this->createMock(AccessDecisionManagerInterface::class),
+        );
+
+        $this->assertTrue($provider->supportsType(TableDataContainerProvider::TYPE_PREFIX.'foobar'));
+        $this->assertFalse($provider->supportsType('foobar'));
+    }
+
+    public function testUpdateIndex(): void
+    {
+        $connection = $this->createInMemorySQLiteConnection(
+            [
+                new Table('tl_content', [
+                    new Column('id', Type::getType(Types::INTEGER)),
+                    new Column('type', Type::getType(Types::STRING)),
+                    new Column('text', Type::getType(Types::STRING)),
+                ]),
+                new Table('tl_news', [
+                    new Column('id', Type::getType(Types::INTEGER)),
+                    new Column('headline', Type::getType(Types::STRING)),
+                    new Column('teaser', Type::getType(Types::STRING)),
+                ]),
+            ],
+            [
+                'tl_content' => [
+                    [
+                        'id' => 1,
+                        'type' => 'text',
+                        'text' => '<p>This is <em>some</em> content.',
+                    ],
+                ],
+                'tl_news' => [
+                    [
+                        'id' => 2,
+                        'headline' => 'This is my news 2!',
+                        'teaser' => 'A great teaser!',
+                    ],
+                    [
+                        'id' => 3,
+                        'headline' => 'This is my news 3!',
+                        'teaser' => 'Another great teaser!',
+                    ],
+                ],
+            ],
+        );
+
+        $framework = $this->mockContaoFramework();
+
+        $fixturesDir = $this->getFixturesDir();
+        $resourceFinder = new ResourceFinder(Path::join($fixturesDir, 'table-data-container-provider'));
+        $locator = new FileLocator(Path::join($fixturesDir, 'table-data-container-provider'));
+
+        $container = $this->getContainerWithContaoConfiguration($fixturesDir);
+
+        $container->set('contao.resource_finder', $resourceFinder);
+        $container->set('contao.resource_locator', $locator);
+        System::setContainer($container);
+
+        $provider = new TableDataContainerProvider(
+            $framework,
+            $resourceFinder,
+            $connection,
+            $this->createMock(RecordLabeler::class),
+            $this->createMock(AccessDecisionManagerInterface::class),
+        );
+
+        $documents = $provider->updateIndex(new UpdateAllProvidersConfig());
+
+        // Sort the documents for deterministic tests
+        /** @var array<Document> $documents */
+        $documents = iterator_to_array($documents);
```

Comment:
lol, yeah what I wanted to do is to document the left `$documents`, not the right :D

Reply:
The inline phpDoc does not seem to be the problem. Even if I delete it, the `Expected parameter of type '\Traversable', `\Contao\CoreBundle\Search\Backend\Document[]' provided` warning remains. Maybe the return type of the `TableDataContainerProvider::updateIndex()` method needs to be adjusted?

Reply:
I don't really know what to do here because I think this is a phpstorm bug, isn't it? `iterable` is a built-in compile time type alias for `array|Traversable` so it is `Traversable`. And phpstan would complain if there was a type issue here 🤔